### PR TITLE
client: Fix wrong pickN generator

### PR DESCRIPTION
**Pull request by exFalso**
Small one

### CODE REVIEW

File: client/src/main/kotlin/net/corda/client/mock/Generator.kt [hunk 1]
```diff
@@ -176,14 +176,15 @@ fun <A : Any> Generator.Companion.replicatePoisson(meanSize: Double, generator:
 fun <A : Any> Generator.Companion.pickOne(list: List<A>) = Generator.intRange(0, list.size - 1).map { list[it] }
 fun <A : Any> Generator.Companion.pickN(number: Int, list: List<A>) = Generator<List<A>> {
     val mask = BitSet(list.size)
-    for (i in 0..Math.min(list.size, number) - 1) {
-        mask[i] = 1
+    val size = Math.min(list.size, number)
+    for (i in 0..size - 1) {
+        mask[i] = true
```
Comment (mnesbit) [inline client/src/main/kotlin/net/corda/client/mock/Generator.kt]: Perhaps just a comment warning of the subtle perils of the old code

File: client/src/main/kotlin/net/corda/client/mock/Generator.kt [hunk 2]
```diff
@@ -176,14 +176,15 @@ fun <A : Any> Generator.Companion.replicatePoisson(meanSize: Double, generator:
...
     }
-    for (i in 0..mask.size() - 1) {
-        val byte = mask[i]
-        val swapIndex = i + it.nextInt(mask.size() - i)
+    for (i in 0..size - 1) {
+        val bit = mask[i]
+        val swapIndex = i + it.nextInt(size - i)
         mask[i] = mask[swapIndex]
-        mask[swapIndex] = byte
+        mask[swapIndex] = bit
```
Comment (adagys) [inline client/src/main/kotlin/net/corda/client/mock/Generator.kt]: Shuffling here doesn't really change anything, no? `mask[0]` to `mask[size-1]` remains `true`. Also could you comment what's the supposed behaviour of `pickN`?

Comment (exFalso) [inline client/src/main/kotlin/net/corda/client/mock/Generator.kt]: Ah you're very right, should be going up to `list.size`!

Comment (adagys) [inline client/src/main/kotlin/net/corda/client/mock/Generator.kt]: Ok, but then it's essentially the same as the previous code, what's the purpose of this change?

Comment (exFalso) [inline client/src/main/kotlin/net/corda/client/mock/Generator.kt]: It's not the same, the problem was with `mask[i] = 1`, which does something completely counterintuitive, i'll add a comment for this

**Comment by exFalso** [comment]
Added small comment and fixed for loop, ball in your court
